source/model/Activator.py

```diff
@@ -7,8 +7,11 @@ class Activator(ABC):
     Concrete activators of a plugin should implement this class
     """
 
-    def __init__(self, id):
-        self.activatorId = id
+    @property
+    @abstractmethod
```

Comment:
Just a question. As all activators require an Id I think it is best that it is not an abstract  method but that activator itself implements the id. This way a lot of duplicate code is removed.
What do you think?